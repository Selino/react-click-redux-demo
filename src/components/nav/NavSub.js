/** @jsx jsx */
import { css, jsx } from "@emotion/core"
import { useState, useEffect } from "react"
import getMenuText from "../../fixtures/MenuText"
import { useTransition, animated } from "react-spring"

function AnimatedTable({ title, desc }) {
  const transitions = useTransition(title, null, {
    from: {
      position: "absolute",
      top: ".5rem",
      opacity: 0,
      marginLeft: -1000,
    },
    enter: { opacity: 1, marginLeft: 0 },
    leave: { opacity: 0, marginLeft: -1000 },
  })

  return transitions.map(
    ({ item, key, props }) =>
      item && (
        <animated.table key={key} style={props}>
          <tbody>
            <tr style={{ verticalAlign: "top" }}>
              <td>{desc}</td>
            </tr>
          </tbody>
        </animated.table>
      )
  )
}

export default function NavSub(props) {
  const [state, setState] = useState({})

  useEffect(() => {
    const subMenuText = getMenuText()
    const title = subMenuText.find((x) => x.link === props.location.pathname)
      .title
    const desc = subMenuText.find((x) => x.link === props.location.pathname)
      .subMenuText
    setState({ title, desc })
  }, [props.location.pathname])

  return (
    <div
      css={css`
        height: 2.5rem;
        margin-top: 66px;
        color: #333333;
        padding: 1rem;
        z-index: 5;
        width: 100%;
      `}
      className='position-fixed'
    >
      <AnimatedTable title={state.title} desc={state.desc} />
    </div>
  )
}
