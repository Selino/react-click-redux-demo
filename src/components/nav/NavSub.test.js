import React from "react"
import { render, screen, fireEvent } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"
import NavSub from "./NavSub"
import { getSubMenuText } from "../../fixtures/SubMenuText"

console.warn = jest.fn()

describe("NavSub", () => {
  test("should match snapshot for each location from the getSubMenuText array", () => {
    getSubMenuText().forEach((item) => {
      const { asFragment } = render(
        <NavSub location={{ pathname: item.url }} />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  test("should match title and desc strings from getSubMenuText array", () => {
    getSubMenuText().forEach((item) => {
      render(<NavSub location={{ pathname: item.url }} />)
      expect(screen.getAllByText(item.title))
    })
  })
})
