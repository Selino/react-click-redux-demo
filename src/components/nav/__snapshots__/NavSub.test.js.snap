// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavSub should match snapshot for each location from the getSubMenuText array 1`] = `
<DocumentFragment>
  <div
    class="position-fixed"
    style="margin-top: 58px; background-color: rgb(51, 51, 51); color: white; padding: 1rem; z-index: 5; width: 100%;"
  >
    <table
      class="react-reveal"
      style="opacity: 1;"
    >
      <tbody>
        <tr
          style="vertical-align: top;"
        >
          <td
            style="color: rgb(255, 213, 0); font-weight: bold; padding-right: 1rem;"
          >
            root
          </td>
          <td>
            root
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`NavSub should match snapshot for each location from the getSubMenuText array 2`] = `
<DocumentFragment>
  <div
    class="position-fixed"
    style="margin-top: 58px; background-color: rgb(51, 51, 51); color: white; padding: 1rem; z-index: 5; width: 100%;"
  >
    <table
      class="react-reveal"
      style="opacity: 1;"
    >
      <tbody>
        <tr
          style="vertical-align: top;"
        >
          <td
            style="color: rgb(255, 213, 0); font-weight: bold; padding-right: 1rem;"
          >
            Home
          </td>
          <td>
            Choose an item from the list below.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`NavSub should match snapshot for each location from the getSubMenuText array 3`] = `
<DocumentFragment>
  <div
    class="position-fixed"
    style="margin-top: 58px; background-color: rgb(51, 51, 51); color: white; padding: 1rem; z-index: 5; width: 100%;"
  >
    <table
      class="react-reveal"
      style="opacity: 1;"
    >
      <tbody>
        <tr
          style="vertical-align: top;"
        >
          <td
            style="color: rgb(255, 213, 0); font-weight: bold; padding-right: 1rem;"
          >
            Redux Counter Prototype
          </td>
          <td>
            An over-engineered counter as a sandbox for learning Redux.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`NavSub should match snapshot for each location from the getSubMenuText array 4`] = `
<DocumentFragment>
  <div
    class="position-fixed"
    style="margin-top: 58px; background-color: rgb(51, 51, 51); color: white; padding: 1rem; z-index: 5; width: 100%;"
  >
    <table
      class="react-reveal"
      style="opacity: 1;"
    >
      <tbody>
        <tr
          style="vertical-align: top;"
        >
          <td
            style="color: rgb(255, 213, 0); font-weight: bold; padding-right: 1rem;"
          >
            Trilliant Executive Summary
          </td>
          <td>
            A live data table with all the bells and whistles.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`NavSub should match snapshot for each location from the getSubMenuText array 5`] = `
<DocumentFragment>
  <div
    class="position-fixed"
    style="margin-top: 58px; background-color: rgb(51, 51, 51); color: white; padding: 1rem; z-index: 5; width: 100%;"
  >
    <table
      class="react-reveal"
      style="opacity: 1;"
    >
      <tbody>
        <tr
          style="vertical-align: top;"
        >
          <td
            style="color: rgb(255, 213, 0); font-weight: bold; padding-right: 1rem;"
          >
            Word Counter
          </td>
          <td>
            Type is some words and watch the count change! Exciting, right?
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`NavSub should match snapshot for each location from the getSubMenuText array 6`] = `
<DocumentFragment>
  <div
    class="position-fixed"
    style="margin-top: 58px; background-color: rgb(51, 51, 51); color: white; padding: 1rem; z-index: 5; width: 100%;"
  >
    <table
      class="react-reveal"
      style="opacity: 1;"
    >
      <tbody>
        <tr
          style="vertical-align: top;"
        >
          <td
            style="color: rgb(255, 213, 0); font-weight: bold; padding-right: 1rem;"
          >
            Gif Caller
          </td>
          <td>
            A Giphy API caller using Async React Hooks.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`NavSub should match snapshot for each location from the getSubMenuText array 7`] = `
<DocumentFragment>
  <div
    class="position-fixed"
    style="margin-top: 58px; background-color: rgb(51, 51, 51); color: white; padding: 1rem; z-index: 5; width: 100%;"
  >
    <table
      class="react-reveal"
      style="opacity: 1;"
    >
      <tbody>
        <tr
          style="vertical-align: top;"
        >
          <td
            style="color: rgb(255, 213, 0); font-weight: bold; padding-right: 1rem;"
          >
            Fizz Buzz
          </td>
          <td>
            A React version of a classic Javascript puzzle.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
